modules/core/src/main/java/org/apache/synapse/mediators/store/MessageStoreMediator.java

```diff
@@ -78,7 +81,14 @@ public boolean mediate(MessageContext synCtx) {
         }
 
         if(synCtx != null) {
-            MessageStore messageStore = synCtx.getConfiguration().getMessageStore(messageStoreName);
+
+            MessageStore messageStore;
+
+            if(messageStoreExp != null)
+                messageStore = synCtx.getConfiguration().getMessageStore(messageStoreExp.stringValueOf(synCtx));
+            else
+                messageStore = synCtx.getConfiguration().getMessageStore(messageStoreName);
```

Comment:
Please add curly brackets. Even though this is syntactically correct, adding brackets improve the readability.

Reply:
done